main.go

```diff
@@ -950,7 +952,7 @@ func splitFlagsFromArgs(all []string) (flags, args []string) {
 	for i := 0; i < len(all); i++ {
 		arg := all[i]
 		if !strings.HasPrefix(arg, "-") {
-			return all[:i], all[i:]
+			return all[:i:i], all[i:]
```

Comment:
Never seen `all[:i:i]` before what does it do, or is it a bug?

Reply:
Tested doesn't do anything other then `[:i]` so please revert.

Reply:
This is on purpose, to limit the slice's capacity to `i` as well. Otherwise, if `i == 3` and `cap(all) == 5`, `append(flags, "foo")` would modify the first element of the returned `args`.

Reply:
The alternative would be to not return sub-slices of `all`, and instead allocate new slices and copy elements into those. That would be more straightforward (no need for the three-index slice expression), but it would also generate more garbage.

Reply:
Ok that makes sense, didn't know that one could do that.